csharp/src/Ice/UriParser.cs

```diff
@@ -23,95 +31,50 @@ internal static class UriParser
             GenericUriParserOptions.IriParsing |
             GenericUriParserOptions.NoUserInfo;
 
-        /// <summary>Checks if a string is an ice or ice+transport URI and not a stringified proxy using the old format.
-        /// </summary>
+        /// <summary>Checks if a string is an ice or ice+transport URI, and not a proxy or endpoint string using the
+        /// ice1 string format.</summary>
         /// <param name="s">The string to check.</param>
         /// <returns>True when the string is most likely an ice or ice+transport URI; otherwise, false.</returns>
         internal static bool IsUri(string s) =>
```

Comment:
Should we allow Spaces in URI like the URI constructor?

```
Uri uri1 = new Uri("http://localhost/test/path%20/file.txt?resource=foo%20bar");
Uri uri2 = new Uri("http://localhost/test/path /file.txt?resource=foo bar");
Console.WriteLine($"URI1 == URI2 {uri1.Equals(uri2)}");
```

```
URI1 == URI2 True
```

The URI class consider both equivalent, maybe it should be the same for endpoints and accept spaces
https://dotnetfiddle.net/Q8walO

Reply:
I updated the check to no longer use spaces. This means that `ice:tcp -p 10000` (a valid proxy in ice1 format) will now be detected as a URI and get parsed (successfully) as an ice2 indirect proxy. A work-around is to convert it into `ice -t:tcp -p 10000`. Obviously not a common occurrence.